app/src/main/java/com/projectit3048c/MainViewModel.kt

```diff
@@ -11,24 +11,16 @@ import com.projectit3048c.service.IFoodService
 
 class MainViewModel(var foodService : IFoodService =  FoodService()) : ViewModel() {
 
-    var foods: MutableLiveData<Food?> = MutableLiveData<Food?>()
-
+    var foods: MutableLiveData<List<Food>> = MutableLiveData<List<Food>>()
 
     fun fetchFoods() {
         viewModelScope.launch {
-            var innerFoods = foodService.fetchFoods()
-            foods.postValue(innerFoods)
+           var innerFoods = foodService.fetchFoods()
+           foods.postValue(innerFoods!!)
```

Comment:
Is the force unwrap (!!) needed here?

That should be avoided.  If nullable, use let{} and ?: instead.